libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -1241,6 +1247,59 @@ int16_t AP_Logger::Write_calc_msg_len(const char *fmt) const
     return len;
 }
 
+// thread for processing IO - in general IO involves a long blocking DMA write to an SPI device
+// and the thread will sleep while this completes preventing other tasks from running, it therefore
+// is necessary to run the IO in it's own thread
+void AP_Logger::io_thread(void)
+{
+    uint32_t last_sd_start_ms = AP_HAL::millis();
+    uint32_t last_run_us = AP_HAL::micros();
+
+    while (true) {
+        uint32_t now = AP_HAL::micros();
+
+        if (now - last_run_us < 1000) {
+            hal.scheduler->delay_microseconds(1000 - (now - last_run_us));
+        } else {
+            // always have some delay
+            hal.scheduler->delay_microseconds(250);
+        }
```

Comment:
```suggestion
        const uint16_t delay_us = MIN(1000 - (now - last_run_us), 1000);
        hal.scheduler->delay_microseconds(delay_us);
```

Reply:
Problem with that is you might delay 1us.  Another `MAX(..., 250);` in there might be a good idea.

Reply:
Fixed